enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,474 @@
+- Add validation to the installer to prevent invalid EIP configurations.
+  For example: Check for unassociated EIPs before passing to CCM.
+
+### Non-Goals
+- Creation of EIPs in AWS.
+- Static IP usage with NLBs for OpenShift API server, DNS, Nat Gateways, LBs, Instances.
+- To assign IPs from a Customer Owned IP (CoIP) Pool when using Outposts.
+- Set default EIPs for user-created IngressControllers.
+
+## Proposal
+This enhancement adds API fields in the installer and the Ingress Config specification
+to set the AWS EIP for the `default` NLB `IngressController`. The cluster ingress operator will then create
+the `default` `IngressController` and get the EIP assigned to the service LoadBalancer with the annotation
+`service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+Note: `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` is a standard [Kubernetes service annotation](https://kubernetes.io/docs/reference/labels-annotations-taints/#service-beta-kubernetes-io-aws-load-balancer-eip-allocations)
+
+### API Extensions
+
+#### Installer Updates
+- The first API extension for setting `eipAllocations` is in the installer [Platform](https://github.com/openshift/installer/blob/master/pkg/types/aws/platform.go) type, where the new field `eipAllocations` is added as an optional field.
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+...
+
+    // eipAllocations contains Elastic IP (EIP) allocations for AWS resources 
+    // within the cluster.
+    //
+    // +optional
+    EIPAllocations *EIPAllocations `json:"eipAllocations,omitempty"`
+}
+
+// EIPAllocations contains Elastic IP (EIP) allocations for AWS resources 
+// within the cluster.
+type EIPAllocations struct {
+    // ingressNetworkLoadBalancer is a list of IDs for Elastic IP (EIP) addresses that
+    // are assigned to the default AWS NLB IngressController.
+    // The following restrictions apply:
+    //
+    // eipAllocations can only be used with external scope, not internal.
+    // An EIP can be allocated to only a single IngressController.
+    // The number of EIP allocations must match the number of subnets that are used for the load balancer.
+	// Each EIP allocation must be unique.
+    // A maximum of 10 EIP allocations are permitted.
+    //
+    // See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html for general
+    // information about configuration, characteristics, and limitations of Elastic IP addresses.
+    //
+    // +openshift:enable:FeatureGate=InstallEIPForDefaultNLBIngressController
+    // +optional
+    // +listType=atomic
+    // +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="eipAllocations cannot contain duplicates"
+    // +kubebuilder:validation:MaxItems=10
+    IngressNetworkLoadBalancer []EIPAllocation `json:"ingressNetworkLoadBalancer"`
```

Comment:
Is `IngressNetworkLoadBalancer` an existing field name? If it's a new name, it doesn't seem suitable to describe a list of EIP allocations.

Reply:
This name was already discussed with the installer team. I am not able to find the thread.